backend/matching_service/controllers/socketController.js

```diff
@@ -114,6 +119,46 @@ class SocketController {
 
       // Logic for collab service: collabService(this.io, roomId) etc
       // this.io.to(roomId).emit(...)
+      console.log("Creating session...");
+      const sessionId = roomId
+
+      const requestBody = {
+        sessionId: sessionId, // ID of the session
+        sessionData: sessionData, // Data related to the session (user info, etc.)
+        questionData: randomQuestion, // Random question to be included in the session
+      };
+      
+      // Log the request body
+      console.log('Request Body:', JSON.stringify(requestBody));
+      
+      const response = await fetch(createsession, {
+        method: "POST",
+        headers: {
+            'Authorization': `Bearer ${token}`, // If using JWT or similar
+            'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(requestBody),
+      });
+    
+      if (response.ok) {
+          const result = await response.json();
+          console.log('Session created:', result);
+      } else {
+          const error = await response.json();
+          console.error('Error creating session:', error);
+      }
+        console.log("Emitting navigateToCollab to both users with session ID:", sessionId);
+        console.log("Session Data:", JSON.stringify(sessionData, null, 2));
+
+        this.emitNavigateToCollab(prevUserSocket, currUserSocket, sessionId, prevUserSessionData, currUserSessionData, randomQuestion);
+
+        await this.removeExistingConnection(prevUserSessionData.uid);
+        await this.removeExistingConnection(currUserSessionData.uid);
+
+        // Disconnect both users' sockets
+        prevUserSocket.disconnect();
+        currUserSocket.disconnect();
```

Comment:
The `await this.removeExisitingConnection` lines already disconnects the sockets, so we can prune the `disconnect()` lines below.